### PR TITLE
Add the necessary changes for rebalance strategies on enterprise

### PR DESCRIPTION
**Pull request by JelteF**
This PR adds the SQL and C changes necessary to support custom rebalance
strategies in the Enterprise version of Citus.

### CODE REVIEW

File: src/backend/distributed/sql/citus--9.1-1--9.2-1.sql [hunk 1]
```diff
@@ -12,3 +12,73 @@ DROP INDEX pg_dist_colocation_configuration_index;
...
+    default_strategy boolean NOT NULL DEFAULT false,
+    shard_cost_function regproc NOT NULL,
+    node_capacity_function regproc NOT NULL,
+    shard_allowed_on_node_function regproc NOT NULL,
+    default_threshold float4 NOT NULL,
+    minimum_threshold float4 NOT NULL DEFAULT 0,
+    UNIQUE(name)
+);
+ALTER TABLE citus.pg_dist_rebalance_strategy SET SCHEMA pg_catalog;
+GRANT SELECT ON pg_catalog.pg_dist_rebalance_strategy TO public;
```
Comment (thanodnl) [inline src/backend/distributed/sql/citus--9.1-1--9.2-1.sql]: @marcocitus I guess it is ok for anyone to read right? can you confirm

Comment (marcocitus) [reply src/backend/distributed/sql/citus--9.1-1--9.2-1.sql]: seems fine, we do it for other tables

**Comment by thanodnl** [comment]
Coverage is a bit wacky, but important lines seem to be covered.

**Comment by JelteF** [comment]
Coverage for the failure cases is done in the enterprise repo.

**Pull request merged**